InternalEngineTests testKeepTranslogAfterGlobalCheckpoint reproducible failure

`InternalEngineTests#testKeepTranslogAfterGlobalCheckpoint` failed on 7.x in CI and the failure reproduced locally for me.

```
./gradlew ':server:test' --tests "org.elasticsearch.index.engine.InternalEngineTests.testKeepTranslogAfterGlobalCheckpoint" -Dtests.seed=8FAEE298979A1A1F -Dtests.security.manager=true -Dtests.locale=uk -Dtests.timezone=Europe/Istanbul -Dcompiler.java=13
```

```
org.elasticsearch.index.engine.InternalEngineTests > testKeepTranslogAfterGlobalCheckpoint FAILED
    java.lang.AssertionError: 
    Expected: a value less than or equal to <32L>
         but: <33L> was greater than <32L>
        at __randomizedtesting.SeedInfo.seed([8FAEE298979A1A1F:92ADF96D18B4F38]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.engine.InternalEngineTests.testKeepTranslogAfterGlobalCheckpoint(InternalEngineTests.java:4882)
```

Build scan: https://gradle-enterprise.elastic.co/s/yxgoiqxlkrs5y

--- a/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
@@ -4828,7 +4828,6 @@ public class InternalEngineTests extends EngineTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/53505")
     public void testKeepTranslogAfterGlobalCheckpoint() throws Exception {
         IOUtils.close(engine, store);
         final IndexSettings indexSettings = new IndexSettings(defaultSettings.getIndexMetaData(), defaultSettings.getNodeSettings(),
@@ -4870,8 +4869,8 @@ public class InternalEngineTests extends EngineTestCase {
                     engine.syncTranslog();
                 }
                 if (frequently()) {
-                    final long lastSyncedGlobalCheckpoint = Translog.readGlobalCheckpoint(translogPath, translogUUID);
                     engine.flush(randomBoolean(), true);
+                    final long lastSyncedGlobalCheckpoint = Translog.readGlobalCheckpoint(translogPath, translogUUID);
                     final List<IndexCommit> commits = DirectoryReader.listCommits(store.directory());
                     // Keep only one safe commit as the oldest commit.
                     final IndexCommit safeCommit = commits.get(0);

